### PR TITLE
chore: add APIScan

### PR DESCRIPTION
**Pull request by rzhao271**

### CODE REVIEW

**Comment by rzhao271** [comment]
Verification build https://dev.azure.com/vscode/ripgrep-prebuilt/_build/results?buildId=115783&view=results

**Comment by rzhao271** [comment]
APIScan doesn't support Rust, but this pipeline also confirms that Rust nightly with -Zprofile isn't supported.

Verification build https://dev.azure.com/vscode/ripgrep-prebuilt/_build/results?buildId=115903&view=results

File: build/windows-apiscan.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# Based on https://github.com/BurntSushi/ripgrep/blob/master/appveyor.yml
+
+parameters:
+  target: ''
+  rust_version: 'nightly'
```
Comment (rzhao271) [inline build/windows-apiscan.yml:5]: Changed from windows.yml

File: build/windows-apiscan.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    rustup target add $env:TARGET
+  env:
+    RUST_VERSION: ${{ parameters.rust_version }}
+    TARGET: ${{ parameters.target }}
+    reason: variables['Build.Reason']
+  displayName: Install Rust
+
+- powershell: |
+    $env:RUSTFLAGS='-C target-feature=+crt-static -Zprofile'
+    $env:CARGO_INCREMENTAL='0'
```
Comment (rzhao271) [inline build/windows-apiscan.yml:37]: These env vars differ from windows.yml

File: build/windows-apiscan.yml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    cargo build --target $env:TARGET --features pcre2
```
Comment (rzhao271) [inline build/windows-apiscan.yml:38]: Changed from windows.yml (not release build)

**Comment by rzhao271** [comment]
All green now https://dev.azure.com/vscode/ripgrep-prebuilt/_build/results?buildId=115910&view=results

**Review by sbatten — APPROVED** [review]

**Pull request merged**